cfe_internal/enterprise/federation/federation.cf

```diff
@@ -0,0 +1,210 @@
+      home_dir => "$(home)";
+  files:
+    "$(create_files)"
+      create => "true";
+    "$(home)/."
+      depth_search => recurse_with_base("inf"),
+      file_select => all,
+      perms => mo("600", "$(user)");
+  commands:
+    selinux_enabled.missing_ssh_context::
+      "restorecon -F $(home)/.ssh/authorized_keys"
+        contain => in_shell;
+}
+
+
+bundle agent federation_manage_files
+{
+  files:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "/opt/cfengine/federation/."
+        create => "true",
+        perms => mo("600", "root");
+      "/opt/cfengine/federation/cfapache/."
+        create => "true",
+        perms => mo("600", "cfapache");
+      "/opt/cfengine/federation/cfapache/."
+        depth_search => recurse("inf"),
+        file_select => all,
+        perms => mo("600", "cfapache");
+    federation_enabled::
+      "$(federation_config.bin_dir)/."
+        create => "true",
+        perms => mo("600", "root");
+    federation_am_superhub::
+      "$(federation_config.federation_dir)/superhub/."
+        create => "true",
+        perms => mo("600", "root");
+      "$(federation_config.federation_dir)/superhub/drop/."
+        create => "true",
+        perms => mo("600", "root");
+      "$(federation_config.federation_dir)/superhub/import/."
+        create => "true",
+        perms => mo("600", "root");
+    federation_am_feeder::
+      "$(federation_config.federation_dir)/fedhub/."
+        create => "true",
+        perms => mo("600", "root");
+      "$(federation_config.federation_dir)/fedhub/dump/."
+        create => "true",
+        perms => mo("600", "root");
+      "$(federation_config.federation_dir)/fedhub/transport/."
+        create => "true",
+        perms => mo("600", "root");
+
+    # TODO: Render these from templates:
+    federation_am_feeder::
+      "$(federation_config.bin_dir)/dump.sh"
+        create => "true",
+        perms => mo("600", "root");
+    federation_am_transporter::
+      "$(federation_config.bin_dir)/transport.sh"
+        create => "true",
+        perms => mo("600", "root");
+    federation_am_superhub::
+      "$(federation_config.bin_dir)/import.sh"
+        create => "true",
+        perms => mo("600", "root");
+}
+
+bundle agent federation_manage_database
+{
+  files:
+    "/var/cfengine/state/pg/data/postgresql.conf"
+      edit_line => insert_lines("shared_buffers=1GB");
+    "/var/cfengine/state/pg/data/postgresql.conf"
+      edit_line => insert_lines("max_locks_per_transaction=4000");
+}
+
+bundle agent federation_manage_dump
+{
+  commands:
+    federation_am_feeder::
+      # TODO: pass correct arguments once script is done
+      "bash $(federation_config.bin_dir)/dump.sh"
+        contain => in_shell;
+}
+
+bundle agent federation_manage_transport
+{
+  commands:
+    federation_am_transporter::
+      # TODO: pass correct arguments once script is done
+      "bash $(federation_config.bin_dir)/transport.sh"
+        contain => in_shell;
```

Comment:
If we don't require the shell to work, then we should execute without it. Should be able to use $(paths.bash) to get the path for bash (from manually maintained paths.cf )

Reply:
I see that paths.cf doesn't have an entry for bash. Maybe we can revisit this when we are adding the scripts?